File: Composite/JavaExample/CompositeMenu/src/Menu.java
import java.util.ArrayList;
import java.util.List;

public class Menu extends MenuComponent{
    private List<MenuComponent> items = new ArrayList<>();

    public Menu(String name) {
        super(name);
    }

    public void add(MenuComponent component) {
        items.add(component);
    }

    @Override
    public void show(int indent) {
        String spacing = "  ".repeat(indent);
        System.out.println(spacing + "[" + name.toUpperCase() + "]");

        for (MenuComponent item : items) {
            item.show(indent + 1);
        }
    }
}

File: Composite/CompositeFiles/CompositeFiles/Composite/Menu.cs
using CompositeFiles.Component;
namespace CompositeFiles.Composite
{
    public class Menu : MenuComponent
    {
    private List<MenuComponent> items = new List<MenuComponent>();

        public Menu(string name) : base(name) { }

        public void Add(MenuComponent component)
        {
            items.Add(component);
        }

        public override void Show()
        {
            Console.WriteLine(Name);

            foreach (var item in items)
            {
                item.Show();
            }
        }
    }
 
}
